lms/static/scripts/frontend_apps/utils/format.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Truncate a URL to be `maxLength` or fewer characters.
+ *
+ * If `url` needs to be shortened, `truncateURL` tries to preserve the most
+ * informative parts of the URL using some simple heuristics.
+ *
+ * If `url` is not a valid URL, it is just shortened to the first `maxLength - 1`
+ * characters.
+ *
+ * @param {string} url
+ * @param {number} maxLength
+ */
+export function truncateURL(url, maxLength) {
+  if (url.length <= maxLength) {
+    return url;
+  }
+
+  // Strip the protocol and return if that is sufficient.
+  const urlWithoutScheme = url.replace(/^[^:]+:\/\//, '');
+  if (urlWithoutScheme.length <= maxLength) {
+    return urlWithoutScheme;
+  }
+
+  // Strip the query string and fragment, then continue removing path segments
```

Comment:
Pedantically, we never actually _strip_ anything, right? We're just not _using_ the query or fragment in the composed string later. Also, I think this comment may be more cogent _after_ the URL parsing code, maybe combined into the commenting I request in a subsequent note?